draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1291,3 +1278,515 @@ To address this, our HashToBase algorithm greedily takes as many bits as
 possible before reducing mod p, in order to smooth out this bias. This is
 preferable to an iterated procedure, such as rejection sampling, since this
 can be hard to reliably implement in constant time.
+
+The running time of each map2curve function is dominated by the cost of
+finite field inversion. Assuming T_i(F) is the time of inversion in field F,
+a rough bound on the running time of each map2curve function is O(T_i(F))
```

Comment:
Thinking about this a bit more, there are a few operations which are similar complexity. Inverses, square root, cube root, quadratic residues, are all implemented in constant time using an exponentiation of approx about p. These would all count towards the running time.

My calculation (in terms of # of field mults):
 - Elligator2 has one sqrt, one inv = ~2log(p)
 - Icart has one inv, one cube root = ~ 2log(p)
 - SWU has one inv, three sqrt, two qrs = ~6log(p)
 - SimpleSWU has one inv, two sqrt, one qr. = ~4log(p)

Reply:
(I think that makes sense, any errors I'm blaming on a fever).

Reply:
Actually, those square roots should all be brought down to 1, our implementation is inefficient.... We should compute the square root after doing a conditional swap on the x values so we only compute it once.

Reply:
#78 opened to solve that.